### ISSUE TITLE
CrabServerWorker failing for every job -siteDB related

### ISSUE THREAD

**Issue by ericvaandering** [dmwm/CRAB2#566]
Original Savannah ticket 67798 reported by None on Sat May 22 03:44:12 2010.

Hi 

The crabserverworker is failing for every job and this seems to be siteDB related.

Please see below.

2010-05-22 10:01:44,607:Traceback (most recent call last):
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 171, in run
    sub_jobs, reqs_jobs, matched, unmatched = self.submissionListCreation(taskObj, newRange)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 746, in submissionListCreation
    requirements.append( self.sched_parameter_Glite(id_job, taskObj) )
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 1015, in sched_parameter_Glite
    ceReq = self.ce_list()
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 1048, in ce_list
    self.log, self.serviceconfig)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/SiteScreening/BlackWhiteListParser.py&quot;, line 255, in **init**
    super(CEBlackWhiteListParser, self).**init**(whiteList, blackList, logger, self.siteDBAPI.cmsNametoCE, dict)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/SiteScreening/BlackWhiteListParser.py&quot;, line 56, in **init**
    self.blacklist = set(self.expandList(blackList))
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/SiteScreening/BlackWhiteListParser.py&quot;, line 73, in expandList
    expandedItem = self.mapper(searchTerm)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/SiteDB/SiteDB.py&quot;, line 106, in cmsNametoCE
    ceList = self.cmsNametoList(cmsName, 'CE', file=file)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/SiteDB/SiteDB.py&quot;, line 126, in cmsNametoList
    theInfo = self.getJSON(&quot;CMSNameto&quot;+kind, file=file, name=cmsName)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/SiteDB/SiteDB.py&quot;, line 70, in getJSON
    f = self.refreshCache(file, callname, args)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/Service.py&quot;, line 158, in refreshCache
    self.getData(cachefile, url, inputdata, encoder, decoder)
  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/Service.py&quot;, line 215, in getData
    raise e
sslerror: The read operation timed out

2010-05-22 10:01:44,611:Sending TTXmlLogging.
2010-05-22 10:01:44,611:Registering information:
{'submittedJobs': None, 'SE-White': None, 'exc': 'Traceback (most recent call last):\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 171, in run\n    sub_jobs, reqs_jobs, matched, unmatched = self.submissionListCreation(taskObj, newRange)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 746, in submissionListCreation\n    requirements.append( self.sched_parameter_Glite(id_job, taskObj) )\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 1015, in sched_parameter_Glite\n    ceReq = self.ce_list()\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/crab-server/CRABSERVER_1_1_1_pre12/lib/CrabServerWorker/FatWorker.py&quot;, line 1048, in ce_list\n    self.log, self.serviceconfig)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/SiteScreening/BlackWhiteListParser.py&quot;, line 255, in **init**\n    super(CEBlackWhiteListParser, self).**init**(whiteList, blackList, logger, self.siteDBAPI.cmsNametoCE, dict)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/SiteScreening/BlackWhiteListParser.py&quot;, line 56, in **init**\n    self.blacklist = set(self.expandList(blackList))\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/SiteScreening/BlackWhiteListParser.py&quot;, line 73, in expandList\n    expandedItem = self.mapper(searchTerm)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/SiteDB/SiteDB.py&quot;, line 106, in cmsNametoCE\n    ceList = self.cmsNametoList(cmsName, \'CE\', file=file)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/SiteDB/SiteDB.py&quot;, line 126, in cmsNametoList\n    theInfo = self.getJSON(&quot;CMSNameto&quot;+kind, file=file, name=cmsName)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/SiteDB/SiteDB.py&quot;, line 70, in getJSON\n    f = self.refreshCache(file, callname, args)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/Service.py&quot;, line 158, in refreshCache\n    self.getData(cachefile, url, inputdata, encoder, decoder)\n  File &quot;/home/crab/sw_area/slc5_amd64_gcc434/cms/wmcore/WMCORE_0_1_1_pre21-cmp2/lib/WMCore/Services/Service.py&quot;, line 215, in getData\n    raise e\nsslerror: The read operation timed out\n', 'skippedJobs': None, 'error': 'WorkerError worker_1. Task jfernan2_Trees_QCD_Pt-15_7TeV-pythia6_1LepPt1GeV_noHLT_2_f3l7x5. listMatch.', 'reason': 'Failure in pre-submission init', 'SE-Black': None, 'unmatchedJobs': None, 'range': '[12]', 'CE-White': None, 'time': None, 'notSubmittedJobs': None, 'ev': 'Submission', 'CE-Black': &quot;['T0', 'T1']&quot;}

**Comment by ericvaandering** [comment]
Comment by mcinquil on Sat May 22 05:36:40 2010

It is a problem with siteDB:
https://hypernews.cern.ch/HyperNews/CMS/get/cmpAnnounce/405.html

Is the CrabServerWorker marking the jobs as not submitted? Or the component crashes?

Thanks

**Comment by ericvaandering** [comment]
Comment by metson on Sat May 22 07:48:30 2010

Hi, 
See &lt;http://savannah.cern.ch/support/?114625&gt; I think this is resolved now...
Cheers
Simon

**Comment by ericvaandering** [comment]
Comment by farinafa on Mon May 24 03:32:20 2010

Was an Operation Issue, not development.

I wonder why the ticket has been opened even before a deep insight of the problem. In addition the printout states clearly that it is a matter of network latencies.

Fabio

**Comment by ericvaandering** [comment]
Closed by farinafa on Mon May 24 03:32:20 2010

**Comment by ericvaandering** [comment]
Comment by spadhi on Tue May 25 04:43:44 2010

Hi,

Just to close this thread. Fabio and Sanjay talked about
this in details.
1. There was a miscommunication. The ticket was supposed
   to be for the cms-service-webtools. I just do not know how 
   to send them email without using savannah.
2. In case of siteDB failure (as was during the lastweekend)
   we should take a lesson from this (nothing is perfect). I really 
   like the conclusion we came through. The crabserver worker
   can retrieve the time(n-1) local cachefile and submit the task. The update cache method via WMCore should not be responsible 
   for a valid task/job to fail. I opened a savannah for Fabio (thanks!)
   
   &lt;http://savannah.cern.ch/bugs/?67881&gt;
3. In case there are no cache files and there is a network problem, there should be a means to inform dashboard the job is
   in CREATED or FAILED state. 
   
   &lt;http://savannah.cern.ch/bugs/?67882&gt;

Best, Sanjay
